app/gui/src/main/scala/org/bitcoins/gui/dlc/dialog/CreateDLCOfferDialog.scala

```diff
@@ -434,99 +469,100 @@ class CreateDLCOfferDialog extends Logging {
           throw new RuntimeException(
             s"SignedDigitDecompositionEventDescriptors are not supported yet")
       }
-      dialog.dialogPane().getScene.getWindow.sizeToScene()
-    }
-
-    dialog.dialogPane().content = new ScrollPane {
-      content = vbox
+      if (dialog != null) dialog.dialogPane().getScene.getWindow.sizeToScene()
     }
 
-    def getOracleInfo: Option[OracleInfo] = {
-      Try(
-        OracleAnnouncementV0TLV.fromHex(
-          announcementOrContractInfoTF.text.value)) match {
-        case Failure(_) =>
-          Try(
-            ContractInfoV0TLV.fromHex(
-              announcementOrContractInfoTF.text.value)) match {
-            case Failure(_) => None
-            case Success(contractInfo) =>
-              Some(OracleInfo.fromTLV(contractInfo.oracleInfo))
-          }
-        case Success(announcement) => Some(SingleOracleInfo(announcement))
+    if (announcement != null) {
+      onAnnouncementEntered(announcement, contractInfoOpt)
+      contractInfoOpt match {
+        case Some(_) =>
+          announcementOrContractInfoTF.text = contractInfoOpt.get.hex
+        case None => announcementOrContractInfoTF.text = announcement.hex
       }
+    } else if (initialContract.nonEmpty) {
+      announcementOrContractInfoTF.text = initialContract
+      onAnnouncementKeyTyped()
     }
 
-    // When the OK button is clicked, convert the result to a CreateDLCOffer.
-    dialog.resultConverter = dialogButton =>
-      if (dialogButton == ButtonType.OK) {
-        val oracleInfo = getOracleInfo.get
-
-        val collateralLong =
-          numberFormatter.parse(collateralTF.text.value).longValue()
-        val collateral = Satoshis(collateralLong)
+    vbox
+  }
 
-        val feeRateStr = feeRateTF.text.value
+  def getOracleInfo: Option[OracleInfo] = {
+    Try(
+      OracleAnnouncementV0TLV.fromHex(
```

Comment:
Just so you know, we provide methods that return a `Try[]`. In this case, you should be using `OracleAnnouncementV0TLV.fromHexT`. The `T` indicates that this method returns a `Try`. The suffix `Opt` means we are returning `Option[]`. The suffix `F` means we are returning a `Future`.

Reply:
https://github.com/bitcoin-s/bitcoin-s/blob/8f958bf5d9957438d703819a087d87813f3edd89/crypto/src/main/scala/org/bitcoins/crypto/Factory.scala#L25